s3tests/functional/__init__.py

```diff
@@ -364,6 +381,15 @@ def setup():
     # really fail.
     nuke_prefixed_buckets(prefix=prefix)
 
+def parse_hostport(s):
+    out = s.rsplit(":", 1)
+    try:
+        out[1] = int(out[1])
+    except (IndexError, ValueError):
+        # couldn't parse the last component as a port, so let's
+        # assume there isn't a port.
+        out = (s, None)
+    return out
```

Comment:
Is it possible to use a standard library function for parsing host/port by chance?

Reply:
Yes! I managed to get `urlparse()` to work with this!